File: rudder-sdk-core/src/main/java/com/rudderstack/android/sdk/core/RudderScreenInfo.java
package com.rudderstack.android.sdk.core;

import android.app.Application;
import android.content.Context;
import android.util.DisplayMetrics;
import android.view.Display;
import android.view.WindowManager;
import com.google.gson.annotations.SerializedName;

class RudderScreenInfo {
    @SerializedName("density")
    private int density;
    @SerializedName("width")
    private int width;
    @SerializedName("height")
    private int height;

    RudderScreenInfo(Application application) {
        WindowManager manager = (WindowManager) application.getSystemService(Context.WINDOW_SERVICE);
        if (manager != null) {
            Display display = manager.getDefaultDisplay();
            DisplayMetrics displayMetrics = new DisplayMetrics();
            display.getMetrics(displayMetrics);

            this.density = displayMetrics.densityDpi;
            this.height = displayMetrics.heightPixels;
            this.width = displayMetrics.widthPixels;
        }
    }
}

File: models/src/main/java/com/rudderstack/android/models/android/RudderScreenInfo.kt
package com.rudderstack.android.models.android

import com.fasterxml.jackson.annotation.JsonProperty
import com.google.gson.annotations.SerializedName
import com.squareup.moshi.Json


internal class RudderScreenInfo (
    @SerializedName("density")
    @JsonProperty("density")
    @Json(name = "density")
    private val density : Int = 0,

    @SerializedName("width")
    @JsonProperty("width")
    @Json(name = "width")
    private var width : Int= 0,

    @SerializedName("height")
    @JsonProperty("height")
    @Json(name = "height")
    private var height : Int = 0){

    /*init {
        val manager: WindowManager =
            application.getSystemService(Context.WINDOW_SERVICE) as WindowManager
        if (manager != null) {
            val display: Display = manager.getDefaultDisplay()
            val displayMetrics = DisplayMetrics()
            display.getMetrics(displayMetrics)
            density = displayMetrics.densityDpi
            height = displayMetrics.heightPixels
            width = displayMetrics.widthPixels
        }
    }*/
}